setuptools/_entry_points.py

```diff
@@ -0,0 +1,79 @@
+import functools
+import operator
+import itertools
+
+from .extern.jaraco.text import yield_lines
+from .extern.jaraco.functools import pass_none
+from ._importlib import metadata
+from ._itertools import ensure_unique
+
+
+def ensure_valid(ep):
+    """
+    Exercise one of the dynamic properties to trigger
+    the pattern match.
+    """
+    ep.extras
+    return ep
+
+
+def load_group(value, group):
+    """
+    Given a value of an entry point or series of entry points,
+    return each as an EntryPoint.
+    """
+    # normalize to a single sequence of lines
+    lines = yield_lines(value)
+    text = f'[{group}]\n' + '\n'.join(lines)
+    return metadata.EntryPoints._from_text(text)
+
+
+def by_group_and_name(ep):
+    return ep.group, ep.name
+
+
+def validate(eps: metadata.EntryPoints):
+    """
+    Ensure entry points are unique by group and name and validate the pattern.
+    """
+    for ep in ensure_unique(eps, key=by_group_and_name):
+        # exercise one of the dynamic properties to trigger validation
+        ep.extras
```

Comment:
Maybe consider reusing the helper?
```suggestion
        ensure_valid(ep)
```

Reply:
Thanks for flagging this, as I'd lost track of it.

You're right. I was noticing that `ensure_valid` wasn't used anywhere. Some consolidation is called for. If I was going to invoke `ensure_valid`, I'd also want to remove the preceding comment, which becomes redundant, so I'll figure something out separately.

Reply:
Fixed in f4b0e3ac5.